aurelia-bootstrap-select/src/abp-select.js

```diff
@@ -394,10 +394,10 @@ export class AbpSelectCustomElement {
   renderPreSelection() {
     let newValue = this._originalSelectedIndexes || this._originalSelectedObjects;
     let selection = this.findItems(this.collection, newValue, this.objectKey);
-    if (this.isEmptySelection(selection)) {
+    if (this.isEmptySelection(selection) && !this.util.parseBool(this.emptyOnNull) && !this.multiple) {
       this.selectedValue = (this.util.isObject(this.collection[0]) ? this.collection[0][this.objectKey] : this.collection[0]);
       this.selectedItem = this.collection[0];
-    } else {
+    } else if (!this.isEmptySelection(selection)) {
```

Comment:
not sure about this line change, if there's nothing selected then `selectedValue` and `selectedItem` might contain the previous values. I'm just reading the code, so I might be wrong in my assumptions

Reply:
Yeah, good call.